source/material_model/rheology/visco_plastic.cc

```diff
@@ -346,7 +388,12 @@ namespace aspect
                                                                MaterialModel::MaterialUtilities::PhaseUtilities::logarithmic
                                                              );
             output_parameters.composition_viscosities[j] = std::min(std::max(viscosity_yield, minimum_viscosity_for_composition), maximum_viscosity_for_composition);
+
+            // TODO: Assert that:
```

Comment:
Do we need to solve this TODO? If you want to assert that, you want to include a tolerance, e.g. `Assert (std::abs(sum - part_1 - part_2) < std::numeric_limits<double>::epsilon() * 100 * sum, ExcMessage(...))` or whatever tolerance is reasonable.